test/e2e/locality_lb_test.go

```diff
@@ -0,0 +1,299 @@
+        kubernetes.io/hostname: kmesh-testing-control-plane
+      tolerations:
+      - key: "node-role.kubernetes.io/control-plane"
+        operator: "Exists"
+        effect: "NoSchedule"
+`
+		if err := applyManifest(ns, depRemote); err != nil {
+			t.Fatalf("Failed to deploy remote instance (dep2): %v", err)
+		}
+
+		// Deploy a sleep client on the worker node.
+		clientDep := `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sleep
+  labels:
+    app: sleep
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sleep
+  template:
+    metadata:
+      labels:
+        app: sleep
+    spec:
+      containers:
+      - name: sleep
+        image: curlimages/curl
+        command: ["/bin/sleep", "infinity"]
+        imagePullPolicy: IfNotPresent
+      nodeSelector:
+        kubernetes.io/hostname: kmesh-testing-worker
+`
+		if err := applyManifest(ns, clientDep); err != nil {
+			t.Fatalf("Failed to deploy sleep client: %v", err)
+		}
+
+		// Wait for all deployments to be available.
+		deployments := []string{
+			"helloworld-region-zone1-subzone1",
+			"helloworld-region-zone1-subzone2",
+			"sleep",
+		}
+		for _, dep := range deployments {
+			cmd := "kubectl wait --for=condition=available deployment/" + dep + " -n " + ns + " --timeout=120s"
+			if _, err := shell.Execute(true, cmd); err != nil {
+				t.Fatalf("Deployment %s not ready: %v", dep, err)
+			}
+		}
+
+		//  List pods and endpoints after deployments.
+		pods, _ = shell.Execute(true, "kubectl get pods -n "+ns)
+		t.Logf("Pods after deployment in namespace %s:\n%s", ns, pods)
+		endpoints, _ = shell.Execute(true, "kubectl get endpoints helloworld -n "+ns)
+		t.Logf("Endpoints for service helloworld after deployment:\n%s", endpoints)
+
+		// Check DNS resolution from the sleep pod.
+		sleepPod, err := shell.Execute(true, "kubectl get pod -n "+ns+" -l app=sleep -o jsonpath='{.items[0].metadata.name}'")
+		if err != nil || sleepPod == "" {
+			t.Fatalf("Failed to get sleep pod: %v", err)
+		}
+		nslookup, _ := shell.Execute(true, "kubectl exec -n "+ns+" "+sleepPod+" -- nslookup "+fqdn)
+		t.Logf("nslookup output for %s:\n%s", fqdn, nslookup)
+		resolvedIP := extractResolvedIP(nslookup)
+		if resolvedIP == "" {
+			t.Fatalf("Failed to extract resolved IP from nslookup output")
+		}
+		t.Logf("Extracted resolved IP: %s", resolvedIP)
+
+		// Test Locality Preference (PreferClose):
+		t.Log("Testing locality preferred (expect response from region.zone1/subzone1)...")
+		var localResponse string
+		if err := retry.Until(func() bool {
+			t.Logf("Attempting curl request at %s...", time.Now().Format(time.RFC3339))
+			out, execErr := shell.Execute(true,
+				"kubectl exec -n "+ns+" "+sleepPod+" -- curl -v -sSL --resolve "+fqdn+":5000:"+resolvedIP+" http://"+fqdn+":5000/hello")
+			if execErr != nil {
+				t.Logf("Curl error: %v", execErr)
+				return false
+			}
+			t.Logf("Curl output: %s", out)
+			if strings.Contains(out, "region.zone1.subzone1") {
```

Comment:
This test is not right, here you return after checking only `region.zone1.subzone1` response.